hawkular-command-gateway/hawkular-command-gateway-war/src/main/java/org/hawkular/cmdgw/command/ws/WsCommand.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2014-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.cmdgw.command.ws;
+
+import org.hawkular.bus.common.BasicMessage;
+import org.hawkular.bus.common.BasicMessageWithExtraData;
+
+/**
+ * An command to execute on messages coming over a WebSocket.
+ * <p>
+ * Note on thread safety: implementations of this interface must not store any state outside of the
+ * {@link #execute(BasicMessageWithExtraData, WsCommandContext)} method so that the instances can be safely shared
+ * between multiple threads.
+ *
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ * @param <M> the type of the message to process
```

Comment:
Random thought: why do we name type parameters with single letters? We wouldn't accept a code with `void execute(BasicMessageWithExtraData<M> m, WsCommandContext c)`...